sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -473,12 +479,13 @@ namespace Azure { namespace Core { namespace Http {
     }
 
     /**
-     * @brief Add an HTTP header.
+     * @brief Add an HTTP header. Returns 0 if header is successfully added, otherwise, this method
+     * would return the index of the fist invalid char from the header \p name.
      *
      * @param name HTTP header name.
      * @param value HTTP header value.
      */
-    void AddHeader(std::string const& name, std::string const& value);
+    size_t AddHeader(std::string const& name, std::string const& value);
```

Comment:
Why are we returning `size_t` instead of something like `int32_t`?

Reply:
the size of std::string is size_t.  We loop all the size while validating.